src/tool/hpcrun/sample-sources/nvidia.c

```diff
@@ -375,19 +371,12 @@ METHOD_FN(process_event_list, int lush_metrics)
 			    &device_finalizer_shutdown);
 
   // Get control knobs
-  int device_buffer_size =
-    control_knob_value_get_int(HPCRUN_CUDA_DEVICE_BUFFER_SIZE);
+  int device_buffer_size;
+  assert(control_knob_value_get_int("HPCRUN_CUDA_DEVICE_BUFFER_SIZE", &device_buffer_size) == 0);
```

Comment:
putting this in assertions is a bad idea. assertions can get elided if compiled with NDEBUG defined.